src_code/src/controllers/services/Authentication/SignUp.js

```diff
@@ -0,0 +1,163 @@
+        if(!userRole || !firstName || !lastName || !emailAddress || !accountPassword) {
+            return responseBuilder.MissingContent(res);
+        }
+
+        if(userRole.trim() !== "Student" && userRole.trim() !== "Faculty" && userRole.trim() !== "Admin") {
+            return responseBuilder.BadRequest(res, "Invalid user role.")
+        }
+
+        if(firstName.length > 25) {
+            return responseBuilder.BadRequest(res, "First name is longer than 25 characters.");
+        }
+
+        if(lastName.length > 25) {
+            return responseBuilder.BadRequest(res, "Last name is longer than 25 characters.");
+        }
+
+        if(req.studentId && req.studentId.length > 0){
+            /** Verifying if the student id is valid */
+            if(req.studentId.length !== 9 || !studentIDPattern.test(req.studentId)) {
+                return responseBuilder.BadRequest(res, "Invalid student ID.");
+            }
+
+            /** Retrieve user with the sign up student id to check for the uniqueness */
+            const isUserWithRequestStudentId = await dbHelper.GetUserInfoByStudentId(db, studentId.trim());
+            
+            if(isUserWithRequestStudentId && typeof isUserWithRequestStudentId === "string") {
+                return responseBuilder.BadRequest(res, isUserWithRequestStudentId);
+            }
+
+            /** If there is already an user with the sign up student id, return bad request */
+            if(isUserWithRequestStudentId) {
+                return responseBuilder.BadRequest(res, "Student ID is already used.");
+            }
+        }
+
+        /** If accountPassword does not meet the password criteria, return error */
+        if(accountPassword.length < 6) {
+            return responseBuilder.BadRequest(res, "Password must be at least 6 characters long.")
+        }
+
+        /** Check if emailAddress ends with @spu.edu */
+        if(!emailPattern.test(emailAddress.trim())){
+            return responseBuilder.BadRequest(res, "Please enter a valid spu email address.");
+        }
+
+        /** Retrieve user with the sign up email address to check for the uniqueness */
+        const isUserWithRequestEmail = await dbHelper.GetUserInfoByEmailAddress(db, emailAddress.trim());
+
+        if(isUserWithRequestEmail && typeof isUserWithRequestEmail === "string") {
+            return responseBuilder.BadRequest(res, isUserWithRequestEmail);
+        }
+        /** If there is already an user with the sign up email address, return bad request */
+        if(isUserWithRequestEmail) { 
+            return responseBuilder.BadRequest(res,"The email address is already in used.");
+        }
+
+        /** Return null to indicate passing the validation */
+        return null;
+    } catch(error) {
+        /** If an unexpected server error occurs while validating the sign up, return log and error. */
+        console.log("ERROR: An error occurred while signing up: ", error);
```

Comment:
Another console.log for testing?

Reply:
It is for AWS Cloudwatch Log, that will not be appeared on the frontend.